### DISCUSSION TITLE
Search special characters

### DISCUSSION THREAD

**Discussion by AllanOliveiraM** [prisma/prisma#22487] [Q&A]
### Question

Hello Prisma team! How are you all doing?

I've been facing difficulties while performing searches using the ORM when there are special characters or characters used here in Brazil. For example, queries using "ç" do not return data, even when the strings in the database contain the same character. I've researched extensively, but I'm having trouble understanding how to resolve this. I'd like to possibly use Trigram-based searches in some cases, but my knowledge of databases is basic, and I'm struggling to put this into practice. From what I've seen, full-text searches (currently in preview) wouldn't fit my use case. Currently, I'm using PostgreSQL, and the Prisma version is 5.5.2.
I've made several attempts, but none have been successful, and I'd like to know what I'm doing wrong. Can someone help me? I appreciate it in advance.

### How to reproduce (optional)

Use `findMany`, for example, with 'contains', 'equals', 'startsWith'...

Search for words containing characters used in Brazil, such as 'ç'.
For instance:
Search for names containing 'Espaço' (case insensitive).
```js
name: {
  contains: 'Espaço',
  mode: 'insensitive',
},
```

**Comment by ludralph** [comment]
HI @AllanOliveiraM 👋 

Thank you for raising this request. We are doing fine, Thank you.

We do have an open issue for filtering using accent insensitive which is exactly the issue you are currently facing.

A user shared a [workaround](https://github.com/prisma/prisma/issues/16211#issuecomment-1364584082) where they installed the `unaccent` extension in Postgres and used the Prisma's [queryRaw](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries#queryraw) to perform the queries. You can take a look at the issue to see if it helps resolve your issue.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by AllanOliveiraM** [reply]
Thank you @ludralph very much for your response! I took a good look at the suggested solution.
Initially, it would indeed solve the issue. However, if I need the raw queries for all search cases in my application, it becomes somewhat unfeasible in my case. This would somewhat defeat one of the purposes of using the ORM here in my project, so I've decided not to pursue this path for now. Perhaps I might not have another choice at the moment :|

I recently had an idea regarding this matter, what's your opinion on the possibility of saving the string in a duplicated way in the database?

Removing all special characters from the string and saving it in a second column,

{
name String
nameForSearch (improve this name) String

@@index([nameForSearch]) ... (maybe?)
}

And then using this column for searching?

It would likely pose a scalability issue, as we'd need to handle the side effects of updating the name without forgetting this column. That's where abstractions in my app would come into play, so it could be feasible. Storage would also be affected, but I don't think it would be significantly impacted.

If you know of any issues with doing this, I'm open to ideas :)

**Reply by mismosmi** [reply]
you might as well go all the way in this case and use postgres full text search features. I've recently had to go through this and will write a blog post soon-ish to be found on https://farbenmeer.de/en/blog. I'll try to put a link here when it's done.

Essentially, there's a way to generate tsvector data in an additional column through a trigger. postgres will then parse your text in a language-specific fashion. this will allow not only for searching for text with and without such language specific special characters but also independent of things like pluralization.

then you can use to_tsquery to generate a query for those tsvectors. You still have to use raw queries for that as it's only supported by prisma in a very simple form but I've done it several times now and it's not too horrible and yields pretty good results.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
